### PR TITLE
Allow persisting needed properties to additional data response for th…

### PR DESCRIPTION
**Pull request by daliwei**
…e initial transactions

cc @alenad

### CODE REVIEW

File: src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentPluginApi.java [hunk 1]
```diff
@@ -238,6 +238,8 @@
     private static final String PREFIX_THREEDS = "threeds2.";
     private static final String PREFIX_THREEDS_RESPONSE_DATA = "threeds2.threeDS2ResponseData.";
 
+    private static final List<String> ADDITIONAL_PROPERTIES_TO_PERSIST = ImmutableList.of("3ds2.csrf", "3ds2.resUrl", "3ds2.formUrl");
```
Comment (alenad) [inline src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentPluginApi.java]: These are groupon specific

File: src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentPluginApi.java [hunk 2]
```diff
@@ -855,6 +858,17 @@ private PaymentTransactionInfoPlugin executeInitialTransaction(final Transaction
         }
     }
 
+    private void addAdditionalDataFromProperty(final PurchaseResult response,
+                                               final Iterable<PluginProperty> properties) {
+        final Map<String, String> map = new HashMap<>();
```
Comment (alenad) [inline src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentPluginApi.java]: Use ImmutableMap.Builder ?

**Review by alenad — CHANGES_REQUESTED** [review]

File: src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentPluginApi.java (hunk 1 above)
Comment (daliwei) [reply src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentPluginApi.java]: Addressed in 60db2b5 to specify these params in the config file.

File: src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentPluginApi.java (hunk 2 above)
Comment (daliwei) [reply src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentPluginApi.java]: Changed the logic in 60db2b5

**Pull request merged**